File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
    #   r, l = 0, 1

        for r in range(len(numbers) - 1): 
            for l in range(r + 1, len(numbers)):
                if numbers[r] + numbers[l] == target:
                    return [r + 1, l + 1]

        return False

File: Data Structures & Algorithms/car-fleet/submission-0.cpp
#include <ranges>

class Solution {
public:
    int carFleet(const int target, const vector<int> &position, const vector<int> &speed) {
        int fleets = 0;
        map<int, int> cars;
        double latestFleetEta = 0;

        for (size_t i = 0; i < position.size(); ++i) {
            cars[position[i]] = speed[i];
        } 

        for (const auto &[position, speed] : cars | std::views::reverse) {
            double eta = static_cast<double>(target-position)/speed;

            if (eta > latestFleetEta) {
                ++fleets;
                latestFleetEta = eta;
            }
        }
              
        return fleets;
    }
};
